File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n = nums.length;

        int[] preProd = new int[n];
        int[] posProd = new int[n];

        preProd[0] = 1;
        posProd[n-1] = 1;

        for(int i=1; i<n; i++){
            preProd[i] = preProd[i-1] * nums[i-1];
            posProd[n-1-i] = posProd[n-i] * nums[n-i];
        }

        for(int i=0; i<n; i++){
            nums[i] = preProd[i] * posProd[i];
        }

        return nums;
    }
}  

File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int ans = 0;

        unordered_set<int> numSet(nums.begin(), nums.end());

        for(int num : numSet){
            int numToFind = num, count = 1;

            while(numSet.find(numToFind-1)!=numSet.end()){
                count++;
                numToFind--;
            }

            ans = max(ans, count);
        }

        return ans;
    }
};
